src/java.base/share/classes/java/net/SocketOptions.java

```diff
@@ -47,224 +47,167 @@
-     * to the peer (for a Socket or DatagramSocket).  The parameter of
-     * this option is an InetAddress.
-     * <P>
-     * This option <B>must</B> be specified in the constructor.
-     * <P>
-     * Valid for: SocketImpl, DatagramSocketImpl
+     * {@link ServerSocket} or {@link DatagramSocket}), or to specify its return address
+     * to the peer (for a {@link Socket} or {@link DatagramSocket}). The type of this option's
+     * value is an {@link InetAddress}.
      *
      * @see Socket#getLocalAddress
      * @see DatagramSocket#getLocalAddress
      */
-
     @Native public static final int SO_BINDADDR = 0x000F;
 
-    /** Sets SO_REUSEADDR for a socket.  This is used only for MulticastSockets
-     * in java, and it is set by default for MulticastSockets.
-     * <P>
-     * Valid for: DatagramSocketImpl
+    /**
+     * See {@link StandardSocketOptions#SO_REUSEADDR} for description of this socket option.
      */
-
     @Native public static final int SO_REUSEADDR = 0x04;
 
-    /** Sets SO_REUSEPORT for a socket. This option enables and disables
-     *  the ability to have multiple sockets listen to the same address
-     *  and port.
-     * <P>
-     * Valid for: SocketImpl, DatagramSocketImpl
-     *
+    /**
+     * See {@link StandardSocketOptions#SO_REUSEPORT} for description of this socket option.
      * @since 9
-     * @see StandardSocketOptions#SO_REUSEPORT
      */
     @Native public static final int SO_REUSEPORT = 0x0E;
 
     /**
-     * Sets SO_BROADCAST for a socket. This option enables and disables
-     * the ability of the process to send broadcast messages. It is supported
-     * for only datagram sockets and only on networks that support
-     * the concept of a broadcast message (e.g. Ethernet, token ring, etc.),
-     * and it is set by default for DatagramSockets.
+     * See {@link StandardSocketOptions#SO_BROADCAST} for description of this socket option.
      * @since 1.4
      */
-
     @Native public static final int SO_BROADCAST = 0x0020;
 
-    /** Set which outgoing interface on which to send multicast packets.
-     * Useful on hosts with multiple network interfaces, where applications
-     * want to use other than the system default.  Takes/returns an InetAddress.
-     * <P>
-     * Valid for Multicast: DatagramSocketImpl
+    /**
+     * See {@link StandardSocketOptions#IP_MULTICAST_IF} for description of this socket option.
      *
      * @see MulticastSocket#setInterface(InetAddress)
      * @see MulticastSocket#getInterface()
      */
-
     @Native public static final int IP_MULTICAST_IF = 0x10;
 
-    /** Same as above. This option is introduced so that the behaviour
-     *  with IP_MULTICAST_IF will be kept the same as before, while
-     *  this new option can support setting outgoing interfaces with either
-     *  IPv4 and IPv6 addresses.
+    /**
+     * Set the outgoing interface on which to send multicast packets. Useful on hosts with
```

Comment:
This option is used to both set and get the outgoing network interface.

Reply:
Updated the PR to state it can be used to both set and get that option.